tests/phpunit/tests/user/queryCache.php

```diff
@@ -0,0 +1,778 @@
+		$user->set_role( 'editor' );
+
+		$q2 = new WP_User_Query(
+			array(
+				'role' => 'author',
+			)
+		);
+
+		$found = wp_list_pluck( $q2->get_results(), 'ID' );
+		$this->assertNotContains( $user_id, $found, 'Expected not to find author in returned values.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_delete_user() {
+		$user_id = self::factory()->user->create();
+
+		$q1 = new WP_User_Query(
+			array(
+				'include' => array( $user_id ),
+			)
+		);
+
+		$found    = wp_list_pluck( $q1->get_results(), 'ID' );
+		$expected = array( $user_id );
+
+		$this->assertSameSets( $expected, $found, 'Find author in returned values' );
+
+		wp_delete_user( $user_id );
+
+		$q2 = new WP_User_Query(
+			array(
+				'include' => array( $user_id ),
+			)
+		);
+
+		$found = wp_list_pluck( $q2->get_results(), 'ID' );
+		$this->assertNotContains( $user_id, $found, 'Expected not to find author in returned values.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_do_not_cache() {
+		$user_id = self::factory()->user->create();
+
+		$args = array(
+			'fields'  => array(
+				'user_login',
+				'user_nicename',
+				'user_email',
+				'user_url',
+				'user_status',
+				'display_name',
+			),
+			'include' => array( $user_id ),
+		);
+
+		$q1       = new WP_User_Query( $args );
+		$found1   = $q1->get_results();
+		$callback = static function( $user ) {
+			return (array) $user;
+		};
+
+		$found1 = array_map( $callback, $found1 );
+
+		$queries_before = get_num_queries();
+		$q2             = new WP_User_Query( $args );
+		$found2         = $q2->get_results();
+		$found2         = array_map( $callback, $found2 );
+		$queries_after  = get_num_queries();
+
+		$this->assertSame( $queries_after - $queries_before, 2, 'Ensure that query is not cached' );
+		$this->assertSameSets( $found1, $found2, 'Expected results to match.', 'Ensure that to results match' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_update_user() {
+		$user_id = self::factory()->user->create();
+
+		wp_update_user(
+			array(
+				'ID'            => $user_id,
+				'user_nicename' => 'paul',
+			)
+		);
+
+		$args = array(
+			'nicename__in' => array( 'paul' ),
+		);
+
+		$q1 = new WP_User_Query( $args );
+
+		$found    = wp_list_pluck( $q1->get_results(), 'ID' );
+		$expected = array( $user_id );
+
+		$this->assertSameSets( $expected, $found, 'Find author in returned values' );
+
+		wp_update_user(
+			array(
+				'ID'            => $user_id,
+				'user_nicename' => 'linda',
+			)
+		);
+
+		$q2 = new WP_User_Query( $args );
+
+		$found = wp_list_pluck( $q2->get_results(), 'ID' );
+		$this->assertNotContains( $user_id, $found, 'Expected not to find author in returned values.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_create_user() {
+		$user_id = self::factory()->user->create();
```

Comment:
As above, any reason you can't use the shared fixtures?

Reply:
Fixed https://github.com/WordPress/wordpress-develop/pull/3544/commits/5a5773cbc8b88fe5d0b9f1361c3485fe498e66da